### ISSUE TITLE
How to handle ranges in a dynamic grouped/sorted columns chart?

### ISSUE THREAD

**Issue by DmitryOganezov** [amcharts/amcharts5#202]
I'm trying to make a "dynamic" grouped/sorted XY chart with ranges.
It's based on this demo:
https://www.amcharts.com/demos/grouped-and-sorted-columns/

I added a few buttons to dynamically rearrange chart data by different fields. 
Everything works as expected, except for the xAxis Ranges.  I can sort data and update xAxis and series dynamically, but all the previous ranges always show up together with the new ones.

What I've tried so far:
1) xAxis.axisRanges.clear(); // - this doesn't work as expected
2) updating axes, series, and ranges in various orders.

So I guess my question is "how to update ranges in a dynamic grouped/sorted columns chart without recreating the entire chart?"

Link on what've got so far:
https://codepen.io/DmitryO/pen/wvrQwNw

![image](https://user-images.githubusercontent.com/12894670/149016095-d7afdb6e-a3da-4a5d-bcac-941c934056a7.png)

**Comment by zeroin** [comment]
To properly dispose axis ranges, do the following:
```
  xAxis.axisRanges.each((range)=>{
    xAxis.removeDispose(range)
  })
```
We will discuss the `clear` issue and will let you know if/when it is fixed

**Comment by DmitryOganezov** [comment]
Thank a lot! 
xAxis.removeDispose(range) works just fine (couldn't find it in the docs, though).

**Comment by martynasma** [comment]
Fixed in [5.1.0](https://github.com/amcharts/amcharts5/releases/tag/5.1.0).

## [5.1.0] - 2022-01-20

### Added
- New `Root` setting: `timezone`. E.g. `"America/Vancouver"`, `"Australia/Sydney"`, `"UTC"`.
- New export CSV option: `addBOM` (default: `true`). Indicates whether to add BOM characters at start of CSV output so that data viewer (e.g. Excel) knows we're exporting UTF-8 data.
- New methods for `Adapters`: `remove(key)`, `disable(key)`, and `enable(key)`.
- `GaplessDateAxis` added. It's a version of `DateAxis` that removes data-less intervals from its scale. `CategoryAxis` is being deprecated in favor of `GaplessDateAxis`. [More info](https://www.amcharts.com/docs/v5/charts/xy-chart/axes/gapless-date-axis/).
- Timezone support added. [More info](https://www.amcharts.com/docs/v5/getting-started/root-element/#Time_zone).
- New `Component` method: `markDirtyValues()`. Will re-evaluate elements of the component that are data-reliant, such as axis labels.

### Fixed
- Clearing axis ranges from Axis and Series was not working properly.
- Slice with negative arc was ignoring `cornerRadius` property.
- `groupDataDisabled` on `XYSeries` was not working if set to `true`.
- Axis header was not updating its position if axes were stacked and size of an axis was updated.
- Disposing of charts with a lot of elements was quite slow.
- Series stacks with all-zero values could cause all stacks to disappear on zoom.

[Full change log](https://github.com/amcharts/amcharts5/blob/master/packages/shared/CHANGELOG.md).

[Download options](https://www.amcharts.com/download/).

Make sure you clear your browser cache after upgrading. And feel free to contact us again if you are still experiencing this issue.
